inference/models/gaze/gaze.py

```diff
@@ -0,0 +1,334 @@
+            img_batch = []
+            for j in range(i, min(len(pil_imgs), i + GAZE_MAX_BATCH_SIZE)):
+                img = self._gaze_transformations(pil_imgs[j])
+                img = np.expand_dims(img, axis=0).astype(np.float32)
+                img_batch.append(img)
+
+            img_batch = np.concatenate(img_batch, axis=0)
+            onnx_input_image = {self.gaze_onnx_session.get_inputs()[0].name: img_batch}
+            pitch, yaw = self.gaze_onnx_session.run(None, onnx_input_image)
+
+            for j in range(len(img_batch)):
+                ret.append((pitch[j], yaw[j]))
+
+        return ret
+
+    def _crop_face_img(self, pil_img: Image.Image, face: Detection) -> Image.Image:
+        """Extract facial area in an image.
+
+        Args:
+            pil_img (Image.Image): The PIL image.
+            face (mediapipe.tasks.python.components.containers.detections.Detection): The detected face.
+
+        Returns:
+            Image.Image: Cropped face image.
+        """
+        # extract face area
+        bbox = face.bounding_box
+        x_min = bbox.origin_x
+        y_min = bbox.origin_y
+        x_max = bbox.origin_x + bbox.width
+        y_max = bbox.origin_y + bbox.height
+        face_img = pil_img.crop((x_min, y_min, x_max, y_max))
+        face_img = face_img.resize((224, 224))
+        return face_img
+
+    def detect(self, request: GazeDetectionInferenceRequest) -> List[GazeDetectionInferenceResponse]:
+        """Detect faces and gazes in image(s).
+
+        Args:
+            request (GazeDetectionInferenceRequest): The request object containing the image.
+
+        Returns:
+            List[GazeDetectionInferenceResponse]: The list of response objects containing the faces and corresponding gazes.
+        """
+
+        if isinstance(request.image, list):
+            if len(request.image) > GAZE_MAX_BATCH_SIZE:
+                raise ValueError(
+                    f"The maximum number of images that can be embedded at once is {GAZE_MAX_BATCH_SIZE}"
+                )
+            imgs = request.image
+        else:
+            imgs = [request.image]
+
+        # load pil images
+        t1 = perf_counter()
+        num_img = len(imgs)
+        pil_imgs = [self.load_image(img.type, img.value) for img in imgs]
+        time_load_img = (perf_counter() - t1) / num_img
+
+        # face detection
+        # TODO: face detection for batch
+        t1 = perf_counter()
+        faces = []
+        for pil_img in pil_imgs:
+            if not request.is_cropped_face_image:
+                mp_img = mp.Image(image_format=mp.ImageFormat.SRGB, data=np.asarray(pil_img))
+                faces_per_img = self.face_detector.detect(mp_img).detections
+            else:
+                faces_per_img = [
+                    Detection(bounding_box=BoundingBox(origin_x=0, origin_y=0, width=pil_img[0], height=pil_img[1]),
+                              categories=[Category(score=1.0, category_name="face")], keypoints=[])]
```

Comment:
Is the "score=1.0" then threshold for a face detection? Does it need to be a tunable parameter or is it binary with 1.0  a positive and everything else a negative detection?